### ISSUE TITLE
False positive `undefined-variable` on `if` clause of class-level list comprehension

### ISSUE THREAD

**Issue by OmeGak** [PyCQA/pylint#5389]
### Bug description

```python
class Foo:
    a = ['a', 'b']
    b = ['b', 'c']
    c = [x for x in a if x not in b]
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:4: [E0602(undefined-variable), Foo] Undefined variable 'b'
```

### Expected behavior

No `undefined-variable` error.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.3
Python 3.9.7 (default, Oct 13 2021, 06:45:31)
[Clang 13.0.0 (clang-1300.0.29.3)]
```

**Comment by OmeGak** [comment]
Never mind, this triggers a `NameError` in Python.

**Comment by OmeGak** [comment]
I reported the bug to Python: https://bugs.python.org/issue45899
